lib/utils/profileBatchLoader.ts

```diff
@@ -0,0 +1,279 @@
+import { PUBLIC_RELAYS } from '@/core/env/MetaConfig';
+import { SearchResult } from '@/lib/types/search';
+import { getProfileData } from '@/lib/utils/profileLoadingUtility';
+import { repository } from '@welshman/app';
+import { load } from '@welshman/net';
+
+interface BatchProfileRequest {
+  pubkey: string;
+  relays?: string[];
+  onProfileLoaded?: (profile: SearchResult) => void;
+  onProfileError?: (pubkey: string, error: any) => void;
+}
+
+interface BatchProfileResult {
+  pubkey: string;
+  profile: SearchResult | null;
+  error?: any;
+}
+
+class ProfileBatchLoader {
```

Comment:
Batching should all be handled by `load`. It's simpler to pass bare events around and adapt them to something like `SearchResult` as late as possible.